qutip/tests/test_entropy.py

```diff
@@ -89,6 +89,56 @@ def test_component_selection(self):
         assert abs(qutip.entropy_mutual(dm, [0, 2], [1]) - expect) < 1e-13
 
 
+class TestRelativeEntropy:
+    def test_rho_or_sigma_not_oper(self):
+        rho = qutip.ket("00")
+        sigma = qutip.ket("01")
+        with pytest.raises(TypeError) as exc:
+            qutip.entropy_relative(qutip.ket2dm(rho), sigma)
+        assert str(exc.value) == "Inputs must be density matrices."
+        with pytest.raises(TypeError) as exc:
+            qutip.entropy_relative(rho, qutip.ket2dm(sigma))
+        assert str(exc.value) == "Inputs must be density matrices."
+
+    def test_base_not_2_or_e(self):
+        rho = qutip.ket2dm(qutip.ket("00"))
+        sigma = qutip.ket2dm(qutip.ket("01"))
+        with pytest.raises(ValueError) as exc:
+            qutip.entropy_relative(rho, sigma, base=3)
+        assert str(exc.value) == "Base must be 2 or e."
+
+    def test_infinite_relative_entropy(self):
+        rho = qutip.ket2dm(qutip.ket("00"))
+        sigma = qutip.ket2dm(qutip.ket("01"))
+        assert qutip.entropy_relative(rho, sigma) == np.inf
+
+    def test_base_2_or_e(self):
+        rho = qutip.ket2dm(qutip.ket("00"))
+        sigma = rho + qutip.ket2dm(qutip.ket("01"))
+        sigma = sigma.unit()
+        assert (
+            qutip.entropy_relative(rho, sigma) == pytest.approx(0.69314718)
+        )
+        assert (
+            qutip.entropy_relative(rho, sigma, base=np.e)
+            == pytest.approx(0.69314718)
+        )
+        assert qutip.entropy_relative(rho, sigma, base=2) == pytest.approx(1)
+
+    @pytest.mark.repeat(20)
+    def test_random_dm_with_self(self):
+        rho = qutip.rand_dm(8, pure=False)
+        rel = qutip.entropy_relative(rho, rho)
+        assert abs(rel) < 1e-13
+
+    @pytest.mark.repeat(20)
+    def test_random_rho_sigma(self):
+        rho = qutip.rand_dm(8, pure=False)
+        sigma = qutip.rand_dm(8, pure=False)
+        rel = qutip.entropy_relative(rho, sigma)
+        assert rel >= 0
```

Comment:
(see above comments about enforcing >= 0 output too.)  In my understanding of floating-point, I don't think the function as written _absolutely_ passes this test for all inputs, though failures would be exceedingly rare.

Reply:
Enforcing greater than or equal to zero added, so this should now always work (but isn't a super strong test).

Reply:
Probably still worth keeping in, I guess?  The test would still catch regressions if in the future we change the implementation of the function.